assayist/processor/container_rpm_analyzer.py

```diff
@@ -0,0 +1,110 @@
+# SPDX-License-Identifier: GPL-3.0+
+
+from assayist.processor.base import Analyzer
+from assayist.processor.utils import get_koji_session
+from assayist.processor.logging import log
+from assayist.common.models import content
+
+
+class ContainerRPMAnalyzer(Analyzer):
+    """Analyze the RPMs in a container image layer."""
+
+    def run(self):
+        """Start the container RPM analyzer."""
+        build_info = self.read_metadata_file(self.BUILD_FILE)
+        build_id = build_info['id']
+        if not self.is_container_build(build_info):
+            log.info(f'Skipping build {build_id} because the build is not a container')
+            return
+
+        self.koji_session = get_koji_session()
+        # Create a mapping of arch to archive (container image) so we can easily map to the
+        # parent container archives in a future loop
+        arch_to_archive = {}
+        not_container_msg = 'Skipping archive {0} since it\'s not a container image'
+        for archive in self.koji_session.listArchives(build_id):
+            if not archive['filename'].startswith('docker-image'):
```

Comment:
Would a better test be `if archive['filename']['btype'] == 'image':`?

Reply:
Sure, as long as that's only used for containers

Reply:
I'm going to assume you meant `if archive['btype'] == 'image':`

Reply:
I did mean that second one, yes.

Hmm, I actually think other image types might also use btype 'image' (e.g. build ID 728887), but I think the clincher is: container images will also have an extra.image dict in the metadata.

Annoyingly, the content metadata OSBS submits specifies "docker-image" for the archive type, but Koji seems to map that to "image".

Reply:
Okay, I addressed this. Thank you.